doc/src/tutorial/codegen.rst

```diff
@@ -0,0 +1,516 @@
+    >>> [a.name for a in r.arguments if isinstance(a, InputArgument)]   
+    [x, y]
+
+    >>> [a.name for a in r.arguments if isinstance(a, OutputArgument)]  # doctest: +SKIP
+    [f, out_8598435338387848786]
+    >>> [a.expr for a in r.arguments if isinstance(a, OutputArgument)]  # doctest: +SKIP
+    [1, Matrix([[x, 2]])]
+
+    >>> [a.name for a in r.arguments if isinstance(a, InOutArgument)]
+    [g]
+    >>> [a.expr for a in r.arguments if isinstance(a, InOutArgument)]
+    [g + x]
+
+
+
+Autowrap
+--------
+Autowrap automatically generates code, writes it to disk, compiles it, 
+and imports it into the current session. Main functions of this module are 
+``autowrap``, ``binary_function``, and ``ufuncify``.
+
+It also automatically converts expressions containing ``Indexed`` objects 
+into summations. The classes IndexedBase, Indexed and Idx represent a matrix 
+element M[i, j]. See :ref:`tensor_module` for more on this.
+``autowrap`` creates a wrapper using f2py or Cython and creates a numerical 
+function.
+
+.. note:: The ``autowrap`` callable is not in the sympy namespace automatically,
+   to use it you must first import ``autowrap`` from ``sympy.utilities.autowrap``
+
+
+The callable returned from autowrap() is a binary python function, not a 
+SymPy object. For example::
+
+    >>> from sympy.abc import x, y, z
+    >>> from sympy.utilities.autowrap import autowrap
+    >>> expr = ((x - y + z)**(13)).expand()
+    >>> binary_func = autowrap(expr)    # doctest: +SKIP
+    >>> binary_func(1, 4, 2)    # doctest: +SKIP
+    -1.0
+
+The various flags available with autowrap() help to modify the services 
+provided by the method. 
+The argument ‘tempdir’ tells autowrap to compile the code in a specific 
+directory, and leave the files intact when finished. For instance::
+
+    >>> from sympy.utilities.autowrap import autowrap
+    >>> from sympy.physics.qho_1d import psi_n
+    >>> from sympy import IndexedBase, Idx
+    >>> from sympy import Eq
+    >>> from sympy import symbols
+    >>> x = IndexedBase('x')
+    >>> y = IndexedBase('y')
+    >>> m = symbols('m', integer=True)
+    >>> i = Idx('i', m)
+    >>> a,omega = symbols('a, omega')
+    >>> qho = autowrap(Eq(y[i], psi_n(0, x[i], m, omega)), tempdir='/tmp')  # doctest: +SKIP
+
+Checking the Fortran source code in the directory specified reveals this::
+
+    subroutine autofunc(m, omega, x, y)
+    implicit none
+    INTEGER*4, intent(in) :: m
+    REAL*8, intent(in) :: omega
+    REAL*8, intent(in), dimension(1:m) :: x
+    REAL*8, intent(out), dimension(1:m) :: y
+    INTEGER*4 :: i
+
+    REAL*8, parameter :: hbar = 1.05457162d-34
+    REAL*8, parameter :: pi = 3.14159265358979d0
+    do i = 1, m
+       y(i) = (m*omega)**(1.0d0/4.0d0)*exp(-4.74126166983329d+33*m*omega*x(i &
+             )**2)/(hbar**(1.0d0/4.0d0)*pi**(1.0d0/4.0d0))
+    end do
+
+    end subroutine
+
+Using the argument 'args' along with it changes argument sequence::
```

Comment:
If you use double ticks ``, the literals will print correctly. So args should have two ticks on each side.